src/router.ts

```diff
@@ -17,27 +18,41 @@ export function createRouter<T extends RadixNodeData = RadixNodeData>(
     staticRoutesMap: {},
   };
 
-  const normalizeTrailingSlash = (p) =>
+  const normalizeTrailingSlash = (p: string) =>
     options.strictTrailingSlash ? p : p.replace(/\/$/, "") || "/";
 
   if (options.routes) {
     for (const path in options.routes) {
-      insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
+      Array.isArray(options.routes[path]) &&
+      options.routes[path].length === 2 &&
+      HTTPMethods.includes(options.routes[path][0])
+        ? insert(
+            ctx,
+            normalizeTrailingSlash(path),
+            options.routes[path][1],
+            options.routes[path][0]
+          )
+        : insert(ctx, normalizeTrailingSlash(path), options.routes[path]);
     }
   }
 
   return {
     ctx,
-    // @ts-ignore
-    lookup: (path: string) => lookup(ctx, normalizeTrailingSlash(path)),
-    insert: (path: string, data: any) =>
-      insert(ctx, normalizeTrailingSlash(path), data),
+    // @ts-expect-error - types are not matching
+    lookup: (path: string, options?: { method?: HTTPMethod }) =>
+      lookup(ctx, normalizeTrailingSlash(path), options?.method),
+    insert: (path: string, data: any, options?: { method?: HTTPMethod }) =>
+      insert(ctx, normalizeTrailingSlash(path), data, options?.method),
     remove: (path: string) => remove(ctx, normalizeTrailingSlash(path)),
   };
 }
 
-function lookup(ctx: RadixRouterContext, path: string): MatchedRoute {
-  const staticPathNode = ctx.staticRoutesMap[path];
+function lookup(
+  ctx: RadixRouterContext,
+  path: string,
+  method?: HTTPMethod
+): MatchedRoute {
+  const staticPathNode = ctx.staticRoutesMap[`${method ?? "ALL"} ${path}`];
```

Comment:
As I mentioned in an earlier issue, it was my first implementation with `path?method` however combining both into a string was not safe (even if working). We should create a map of maps (method => path => node)

Reply:
Refactored to 2 dimensional array.